src/main/java/com/plugin/HttpOutput.java

```diff
@@ -66,42 +122,51 @@ public void stop() {
 
 	@Override
 	public void write(List<Message> msgs) throws Exception {
-		for (Message msg : msgs) {
-			writeBuffer(msg.getFields());
-		}
+
+        for (List<Message> partition: ListUtils.partition(msgs, HTTP_BATCH_SIZE)) {
+            List<Map<String, Object>> payload = new FastArrayList();
+            for (Message msg : partition) {
+                payload.add(msg.getFields());
+            }
+            executeRequest(RequestBody.create(
+                    JSON,
+                    this.gson.toJson(payload)
+            ));
+        }
+
 	}
 
 	@Override
 	public void write(Message msg) throws Exception {
-		if (shutdown) {
-			return;
-		}
 
-		writeBuffer(msg.getFields());
+		executeRequest(
+		        RequestBody.create(
+		                JSON,
+		                this.gson.toJson(msg.getFields())
+                )
+        );
 	}
 
-	public void writeBuffer(Map<String, Object> data) throws HttpOutputException {
-		OkHttpClient client = new OkHttpClient();
-		Gson gson = new Gson();
-
-		try {
-			final MediaType JSON = MediaType.parse("application/json; charset=utf-8");
-
-			RequestBody body = RequestBody.create(JSON, gson.toJson(data));
-			Request request = new Request.Builder().url(url).post(body).build();
-			Response response = client.newCall(request).execute();
-			response.close();
-			if (response.code() != 200) {
-				LOG.info("Unexpected HTTP response status " + response.code());
-				throw new HttpOutputException("Unexpected HTTP response status " + response.code());
-			}
-		} catch (IOException e) {
-			LOG.info("Error while posting the stream data to the given API", e);
-            throw new HttpOutputException("Error while posting stream to HTTP.", e);
-		}
+	private void executeRequest(RequestBody requestBody) throws HttpOutputException, IOException {
+        if (shutdown) {
+            return;
+        }
 
+        Request request = new Request.Builder()
+                .url(url)
```

Comment:
To improve the readability, maybe use `this.url` instead of `url`
```suggestion
                .url(this.url)
```